ui/core/components/detailed_results/damage_metrics.tsx

```diff
@@ -0,0 +1,367 @@
+										{
+											name: `Critical Hit`,
+											value: metric.crits,
+											percentage: metric.critPercent,
+										},
+								  ]
+								: []),
+							...(metric.spellType === SpellType.SpellTypeCast
+								? [
+										{
+											name: 'Hit',
+											value: metric.hits,
+											percentage: metric.hitPercent,
+										},
+										{
+											name: `Critical Hit`,
+											value: metric.crits,
+											percentage: metric.critPercent,
+										},
+								  ]
+								: []),
+							...(metric.spellType === SpellType.SpellTypePeriodic
+								? [
+										{
+											name: 'Tick',
+											value: metric.hits,
+											percentage: metric.hitPercent,
+										},
+										{
+											name: `Critical Tick`,
+											value: metric.crits,
+											percentage: metric.critPercent,
+										},
+								  ]
+								: []),
+							{
+								name: 'Glancing Blow',
+								value: metric.glances,
+								percentage: metric.glancePercent,
+							},
+							{
+								name: 'Blocked Blow',
+								value: metric.blocks,
+								percentage: metric.blockPercent,
+							},
+						]}
+					/>;
+				},
+			},
+			{
+				name: 'Avg Hit',
+				getValue: (metric: ActionMetrics) => metric.avgHit,
+				fillCell: (metric: ActionMetrics, cellElem: HTMLElement) => {
+					cellElem.appendChild(<>{formatToCompactNumber(metric.avgHit)}</>);
+
+					<MetricsCombinedTooltipTable
+						tooltipElement={cellElem}
+						tooltipConfig={{
+							onShow: () => {
+								const hideThreatMetrics = !!document.querySelector('.hide-threat-metrics');
+								if (hideThreatMetrics) return false;
+							},
+						}}
+						headerValues={[, 'Amount']}
+						spellSchool={metric.spellSchool}
+						total={metric.avgHitThreat}
+						totalPercentage={100}
+						hasFooter={false}
+						values={[
+							{
+								name: 'Threat',
+								value: metric.avgHitThreat,
+								percentage: 100,
+							},
+						]}
+					/>;
+				},
+			},
+			{
+				name: 'Crit %',
+				getValue: (metric: ActionMetrics) => metric.critPercent,
+				getDisplayString: (metric: ActionMetrics) => formatToPercent(metric.critPercent),
+			},
+			{
+				name: 'Miss %',
+				getValue: (metric: ActionMetrics) => metric.totalMissesPercent,
+				fillCell: (metric: ActionMetrics, cellElem: HTMLElement) => {
+					cellElem.appendChild(<>{formatToPercent(metric.totalMissesPercent)}</>);
+					if (!metric.totalMissesPercent) return;
+
+					<MetricsCombinedTooltipTable
+						tooltipElement={cellElem}
+						spellSchool={metric.spellSchool}
+						total={metric.totalMisses}
+						totalPercentage={metric.totalMissesPercent}
+						values={[
+							{
+								name: 'Miss',
+								value: metric.misses,
+								percentage: metric.missPercent,
+							},
+							{
+								name: 'Parry',
+								value: metric.parries,
+								percentage: metric.parryPercent,
+							},
+							{
+								name: 'Dodge',
+								value: metric.dodges,
```

Comment:
Currently, the displayed Dodge % is completely wrong for bleeds since it gets diluted by the number of ticks. Does this get fixed automatically with your refactor, or are some additional adjustments needed there?

Reply:
No I don't think so, and I'm not sure how to fix this. Wouldn't this require having to track Hits and Ticks (from Dots) separately to actually make it work properly when having a combined spell (with instant + Dot component?)